### PR TITLE
Use customizable app group id

### PR DESCRIPTION
**Pull request by Klabauterman**
With these changes you can define your own app group id instead of `group.BUNDLE_ID` by putting it into the Info.plist. If no `AppGroupId` is found in the Info.plist file it will use the old convention instead.

(I transferred my app from one apple account to another and lost my app group id by that. So the old approach of this repo wasn't working anymore.)

### CODE REVIEW

**Comment by GRBurst** [comment]
@KasemJaffer I highly recommend to look at this and try to merge in into master :-)
There are some use-cases for the group id not being the same as `group.bundle_id`.
I tested this (I had to because of our architecture, I cannot simply use `group.bundle_id`.

@Klabauterman maybe you could squash the commits (at least the last 2 into the example adaptation). Why did you remove the test code?

File: ios/Classes/SwiftReceiveSharingIntentPlugin.swift [hunk 1]
```diff
@@ -76,8 +76,8 @@ public class SwiftReceiveSharingIntentPlugin: NSObject, FlutterPlugin, FlutterSt
     
     private func handleUrl(url: URL?, setInitialData: Bool) -> Bool {
         if let url = url {
-            let appDomain = Bundle.main.bundleIdentifier!
-            let userDefaults = UserDefaults(suiteName: "group.\(appDomain)")
+			let appGroupId = (Bundle.main.object(forInfoDictionaryKey: "AppGroupId") as? String) ?? "group.\(Bundle.main.bundleIdentifier!)"
```
Comment (GRBurst) [inline ios/Classes/SwiftReceiveSharingIntentPlugin.swift:79]: maybe you can fix the intend here?

File: example/README.md [hunk 2]
```diff
@@ -413,7 +415,7 @@ class ShareViewController: SLComposeServiceViewController {
     private func getThumbnailPath(for url: URL) -> URL {
         let fileName = Data(url.lastPathComponent.utf8).base64EncodedString().replacingOccurrences(of: "==", with: "")
         let path = FileManager.default
-            .containerURL(forSecurityApplicationGroupIdentifier: "group.\(hostAppBundleIdentifier)")!
+            .containerURL(forSecurityApplicationGroupIdentifier: this.appGroupId)!
```
Comment (GRBurst) [inline example/README.md:418]: Here, you cannot use `this`. At the other places, `this` was assigned to `self`. As a result, you should either assign `let this = self` or call `self.appGroupId` here.

**Comment by EduardFrank** [comment]
It would be very great, if this PR could be merged.

**Comment by Klabauterman** [comment]
Hey guys, I am sorry, i totally forgot about my PR. I'll check your comments today.

**Comment by Klabauterman** [comment]
@GRBurst commits are squashed, readme corrected, intend fixed, example code changed

**Comment by GRBurst** [comment]
Hey @Klabauterman, I'm still very thankful since I use your PR in my app for quite some time now :+1: 

I think your last commit (0ede807) should be on a different branch since it does not directly relate to the PR.

@KasemJaffer Again I'd really appreciate if you could consider to merge this. It really solves some real world issues. Please make a statement whether or not you are going to merge it and what issues you see towards merging it ;-)

**Comment by KasemJaffer** [comment]
Thank you all for the great work. I'm closing this because i merged this other [PR](https://github.com/KasemJaffer/receive_sharing_intent/pull/170) which has the same functionality. I will be releasing v1.5.0 as soon as possible after i run some testes.

**Pull request closed without merging**